File: mydll.rb
require_relative 'mydll_common'
require_relative 'mydll_command'

include MyDLL

####################################################################################################

if __FILE__ == $PROGRAM_NAME
  MyDLL.load_lib()
  v = MyDLL::Vec2.malloc
  pp v.x, v.y
  v.x = 123
  pp v.x, v.y
  MyDLL::SetVec2(v)
  pp v.x, v.y
  pp v
  MyDLL::PrintVec2(v)
  pp v.x, v.y
  pp v
end

File: mydll.c
#include "mydll.h"
#include <stdio.h>

MYDLL_API void SetVec2(Vec2* v)
{
    v->x = 12.3f;
    v->y = 23.4f;
}

MYDLL_API void SetRect(Rect* r)
{
    r->tl.x = 1.0f;
    r->tl.y = 2.0f;
    r->br.x = 3.0f;
    r->br.y = 4.0f;
}

MYDLL_API void PrintVec2(Vec2 v)
{
    printf("x = %f, y = %f\n", v.x, v.y);
}

MYDLL_API void PrintRect(Rect r)
{
    printf("r.tl = (%f, %f)\n", r.tl.x, r.tl.y);
    printf("r.br = (%f, %f)\n", r.br.x, r.br.y);
}
